frontend/src/components/NavDropdown.tsx

```diff
@@ -0,0 +1,107 @@
+import clsx from "clsx";
+import { noop, toNumber } from "lodash";
+import { IndexType } from "src/types/generalTypes";
+
+import {
+  Dispatch,
+  JSX,
+  MouseEvent,
+  SetStateAction,
+  useEffect,
+  useState,
+} from "react";
+import { Menu, NavDropDownButton } from "@trussworks/react-uswds";
+
+interface NavDropdownProps {
+  activeNavDropdownIndex: IndexType;
+  index: number;
+  isCurrent: boolean;
+  linkText: string;
+  menuItems: JSX.Element[];
+  setActiveNavDropdownIndex: Dispatch<SetStateAction<IndexType>>;
+}
+
+export default function NavDropdown({
+  activeNavDropdownIndex,
+  index,
+  isCurrent,
+  linkText,
+  menuItems,
+  setActiveNavDropdownIndex,
+}: NavDropdownProps): JSX.Element {
+  const [isOpen, setIsOpen] = useState<boolean>(false);
+
+  function eventHandler(e: MouseEvent) {
```

Comment:
this is nice solution, and seems to work perfectly. I couldn't help trying to clean it up just a little tho

```suggestion
  // closes all navs unless the click event falls on a currently closed drop down
  // in which case, set that drop down as active
  function closeNavs(e: MouseEvent) {
    let targetId = null;
    const dropdowns = document.getElementsByName("navDropDownButton");
    for (const dropdown of dropdowns) {
      if (dropdown.contains(e.target as Node)) {
        let targetNode: HTMLElement = e.target as HTMLElement;
        if (targetNode.localName === "span") {
          targetNode = targetNode.parentNode as HTMLElement;
        }
        targetId = toNumber(targetNode.id);
        break;
      }
    }
    setActiveNavDropdownIndex(targetId);
```

Reply:
@doug-s-nava committing this suggestion, would you still like me to try removing the DOM query? Since this is checking to see if the dropdown was clicked via the mouse event, I'm unsure how I'd do it at a higher level.